src/renderers/WebGLRenderer.js

```diff
@@ -2401,10 +2433,18 @@ THREE.WebGLRenderer = function ( parameters ) {
 
 		_lights.directional.length = directionalLength;
 		_lights.spot.length = spotLength;
+		_lights.rectArea.length = rectAreaLength;
 		_lights.point.length = pointLength;
 		_lights.hemi.length = hemiLength;
 
-		_lights.hash = directionalLength + ',' + pointLength + ',' + spotLength + ',' + hemiLength + ',' + _lights.shadows.length;
+		_lights.hash = [
+			directionalLength,
+			pointLength,
+			spotLength,
+			rectAreaLength,
+			hemiLength,
+			_lights.shadows.length
+		].join(',');
```

Comment:
Ah, yeah, I made that change primarily for readability.  Happy to switch it back.  In general, I'd try to promote the practice of adding comments around code choices made for non-intuitive optimization purposes.